### ISSUE TITLE
Not an issue rather a feature request (not sure where to post it)

### ISSUE THREAD

**Issue by raymate** [cybercase/webminidisc#54]
All my early MD units and my full size decks would date and time stamp the recordings, could this be implemented ?

**Comment by cybercase** [comment]
So, if I understand this correctly you'd like the uploaded track to be titled with the timestamp of the upload, is that correct?

e.g.
Like I'm uploading my "recording.wav" and it's automatically titled with "08:33 03-08-2021".

If that's so, we might have a problem with tracks being uploaded at the same time having also the same title (or very close timestamps).
A better alternative could be using the lastModifiedDate attribute of the recording that you're uploading.

Do you mind me asking what's your use case?

**Comment by raymate** [comment]
Hi, thanks for the reply,

No not in the title, this date/time think must be in the header somewhere

(https://www.manualslib.com/manual/163159/Sony-Mds-Jb930.html?page=8)

This features was on all portable recorders (the high end ones) up until about 1999, nearly all full size deck had it well into the 2000

If you search "sony minidisc date recorded" you will find a few references to it.

Im trying to find a video on youtube showing it but no one has filmed the feature.

The only clue I can give you is the very first unit the MZ-1 portable when your using the feature and pressed the "date" button to show the recorded date and time, the time section would also count up in seconds.

So if you have a 3 minute and 10 second song and recorded it at 9am precisely, it would show the date / time as being " 21y 12d 12m am 09:00:00 " at the beginning of the song and at the end it would show " " 21y 12d 12m am 09:03:10 " so as the song is playing the seconds would actually count up in realtime for the recorded date and time, this may have been a bug as all machines after this one simple showed " 21y 12d 12m am 09:00:00 " as being the record date and time stamp.

They also renamed the feature to "date preset"

All purchased pre recorded music albums did not have the date / time flag, it was only audio recorded yourself. 

Not sure if that helps. 

So discs recorded on Webminidisc all have "00y, 00d, 00m and 00,00" when you ask the unit to show date / time stamp.

**Comment by raymate** [comment]
Sorry forgot to answer your question, my use case is its just fun to see when you recorded a disc, I found a bunch of minidisc in a box I recorded years ago, and I was able to check when I made the recording, in this case most was "2003 July 18"

**Comment by cybercase** [comment]
Thanks for all the information, I was not aware that the MiniDisc had such feature.

I've found some old discussion about this topic here: https://lists.fu-berlin.de/pipermail/linux-minidisc/2010-April/msg00016.html

Unfortunately the discussion ends with https://lists.fu-berlin.de/pipermail/linux-minidisc/2010-April/msg00027.html

I've also tried myself to run the suggested command to retrieve the timestamp on my mz-n710, but got the same `NOT_IMPLEMENTED` result.

Therefore I suppose there's no way to retrieve the timestamp via the netmd protocol :(

**Comment by raymate** [comment]
Thank you for looking into the possibility.

I have a couple of full size decks, MDS-JB940 and MDS-JA333ES and they both do time/date stamps, so I looked at the manual for the MDS-JB980 which has NetMD and is basically the MDS-JB940 machine and this MDS-JB980 does not seem to talk about setting up the clock or the date/time for using as time stamp. So I suspect Sony dropped the ability to actually record date/time stamp but the units can still read the stamp if present.

I have one last thing to check somewhere I have some discs that are recorded with the old SonicStage and im going to see what they do in my decks, if they do not show date and time stamps then we know its not going to work. I think it will not be present on the recording but just interested to see.

I just put an old disc into my MZ-N910 MZ-N920 and it does let me see the date/time stamp, its shows as "Rec Date" under the "Display" menu, so the Sony units still have the ability to read it if present. 

Im thinking you might need a disc that has been recorded with the date/time stamp baked in for you to get any results on your MZ-N710

I have a MZ-N810 which is identical to yours unit but has Radio, this is not able to retrieve the date/time stamp info

So perhaps you get an error because your machine cant do it, so perhaps its it possible the NetMD protocol has the ability to read/record it somehow. 

I need to try and get sonic stage working for a test :)

**Comment by raymate** [comment]
But I must also say what you have done with WebMD is saved our community, having this easy transfer method is superb, I use CD ripped AIFF files and i think the MD end being identical to a CD to MD rip, I only use SP mode. 

Only CD to MD on my MDS-JA333ES do sound a little better, but then again it should this had one of the best ATRAC encoders going and all the audiophile grade components seem to make a different. They have just a tad more detail in the final recording compared to a AIFF over WebMD.

WebMD AIFF transferee sound identical to my MDS-JB940

May I ask do you still tinker with this project? I noticed some UI changes some time back

**Comment by cybercase** [comment]
Thanks @raymate, I'm really glad that you like the app and the sound quality.

Yes, I'm still maintaining this project when bugs show up and I can find some spare time.

I've recently received a few great contributions from other developers who really did a great job in adding some new interesting features, especially support for grouping tracks and full-width-titles in #49 .

I'm happy to not be the only dev here; it'll make sure that the app will be supported in the long-term :)

**Comment by raymate** [comment]
Thank you. Thats awesome, pleased you have some help and like you say will keep the project going for many years.

Thats cool im going to look at the grouping.

I did a final test this morning regarding my initial question, setup up on my Mac a virtual PC with Windows XP, install Sonic Stage sent some music to MD and SonicStage indeed does not send / include time stamps. So it seems NetMD ignores this feature.

Before this project I actually had build a specific system to get track names onto the MD, on some full size MiniDisc decks you have the  "Control A1 II" port if you connect this to a specific Sony CD player that can use CD-TEXT and also features the "Control A1 II" or "S-LINK Control A1" then engage synchro CD start it will send the CD-TECT DATA over the A1 port into the MD deck and onto the disc in realtime. Then I went as far as if the CD did not have CD-TEXT I would burn a CDR with the CD-TEXT embedded from iTunes on my Mac and put that into the CD player and transfer.

Now thanks to you I just..... drag and drop :)

All the best and thanks again for your superb support
